### PR TITLE
Limit auto-detected docs in development pods to 100

### PR DESCRIPTION
**Pull request by radex**
See: https://github.com/CocoaPods/CocoaPods/issues/11753

TL;DR:

- In Development Pods, CP will try to auto-detect docs files, and add them to Pods/Development Pods/NAME/Pod/ group in Xcode
- This might be helpful if there's a few files, but trying to add an unlimited number of files (that are just there for a convenience and not required to build the app) to an Xcode project is not a good idea. In my case, I had a `docs-website` folder, containing 30k files in its node_modules.
- Besides bloating Xcodeproj, a very large number of files can cause Xcodeproj's UUID generator to exhaust all system resources and cause a kernel panic ;) Surely that could be improved, but I don't think we should be adding an arbitrary number of docs files to Xcodeproj anyway.

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Limit auto-detected docs in development pods to 100  
  [radex](https://github.com/radex)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
This seems quite arbitrary in my opinion. I would perhaps provide an option in the podfile to globally ignore those or a per-pod configuration:

```ruby
pod 'MyPod', :skip_developer_files => true
```

or something like that.

**Comment by radex** [comment]
I agree that the exact number is arbitrary.

My main counterpoint is that this implicit behavior can break `pod install` in a way that's very hard to debug, or could bloat the Xcode project to the point where it's very unwieldy - without the user even noticing.

For _some_ sufficiently large number of documentation files (100? 1000?), it will be very unlikely that the user actually intended this to happen, especially that it only happens for development pods, and has no impact on the build.

If you still disagree, then I would propose a compromise in which docs are not arbitrarily limited, but a warning is emitted above 100 auto-detected docs so the user knows to use `skip_developer_files: true`

**Comment by philips77** [comment]
I just encountered the issue. Docs generated with DocC have thousands of files and the generated `Pods.xcodeproj/project.pbxproj` is 4.5 MB long, which makes Xcode's new Source Control navigator unusable. Waiting time to even display the changes is counted in minutes on MacBook Pro.
